discovery-infra/test_infra/utils/utils.py

```diff
@@ -789,6 +793,28 @@ def download_iso(image_url, image_path):
             out.write(chunk)
 
 
+def fetch_url(url, timeout=60, max_retries=5):
+    """
+    Returns the response content for the specified URL.
+    Raises an exception in case of any failure.
+    """
+    try:
+        retries = Retry(
+            read=max_retries,
+            status=max_retries,
+            backoff_factor=0.5,
+            status_forcelist=[500])
+        s = Session()
+        s.mount('http://', HTTPAdapter(max_retries=retries))
+
+        log.info(f"Fetching URL: {url}")
+        response = s.get(url, timeout=timeout)
+        response.raise_for_status()
+        return response.content
+    except (RequestException,  HTTPError) as err:
+        raise Exception(f"Failed to GET: {url} ({err})")
```

Comment:
nit: exception msg in lower case (like the other exceptions we have)

Reply:
most exception messages are actually starting with an upper case, so I'll change the added in agent_cluster_install